### ISSUE TITLE
Department Table

### ISSUE THREAD

**Issue by huy-d-nguyen** [ppact/processing#339]
Hi @ayyubibrahimi , up to now, we don't have any department table in wrgl, we just have the `agency` field in some table to keep track on the department relation and we will create new department whenever we have a new agency value. We think that later then we need a table to keep track on the departments so that we can remove the duplicated/redundant departments. How do you think about that? Do you agree with us?

**Comment by ayyubibrahimi** [comment]
Agreed. Thanks for bringing this up! I planned to talk to @rajivsinclair about this at our next meeting. Currently, on the Django side, the duplication/redundant department is an immediate issue because, for example, news articles are attached to one department (i.e., `Evangeline Sheriff's Office`) while personnel/allegations data is attached to the other redundant department (i.e., `Evangeline Parish SO`)

**Comment by ayyubibrahimi** [comment]
Hey @huy-d-nguyen. Here is the department table that I've come up with. The table includes both the correct `slug` and `name` for each department. As far as I know, there are redundant/duplicate departments for all departments, therefore,  all redundant departments should be consolidated based on the values in the table below. Is this what you had in mind? Please feel free to let me know if this isn't helpful.

[department_name_reference_list.csv](https://github.com/ppact/processing/files/9533903/department_name_reference_list.csv)

**Comment by huy-d-nguyen** [comment]
Yes, looks good to us.

**Comment by ayyubibrahimi** [comment]
Awesome. Please let me know if there is anything else that I can do to help.

**Comment by huy-d-nguyen** [comment]
Hi @ayyubibrahimi , have you pushed the department data to wrgl? We tend to import the new Department data to remove the duplication soon. So please let us know when the data of the department table is ready.

**Comment by ayyubibrahimi** [comment]
@huy-d-nguyen here is a link to the [wrgl](https://hub.wrgl.co/@ipno/r/data/refs/heads/agency-reference-list) table

**Comment by huy-d-nguyen** [comment]
Thank you, we will take a look on that.

**Comment by baoea** [comment]
Hi @ayyubibrahimi, could data team provide fields we need such as `city`, `parish` ,`address`,`phone`, `location_map_url`, `location`? Our importing flow depends on the fields data team provides us. Thanks!

**Comment by ayyubibrahimi** [comment]
Sure @baoea. 

@rajivsinclair @lsukaristone - please advise as to how you generate the `location` coordinates currently on Django for each agency.

**Comment by baoea** [comment]
@ayyubibrahimi the current way we get the `location` coordinates is to google them and then manually input those into the admin page.

**Comment by baoea** [comment]
Hi @ayyubibrahimi, could you change the `agency` field in `event` table into `slug` instead of `name`? Thanks!

**Comment by ayyubibrahimi** [comment]
Hey @baoea will look into those changes

**Comment by baoea** [comment]
Thanks @ayyubibrahimi. Since now we have `department` table, we would like to use `slug` as department `id`

**Comment by ayyubibrahimi** [comment]
@baoea - the `department` table has been updated on wrgl with `location` values. 

I don't know of any departments, at the moment, which have values for the `city`, `parish` ,`address`, `phone`, `location_map_url`, fields that you mentioned above. Is this something that should be prioritized, or is the `location` sufficient for the immediate future @baoea @rajivsinclair @lsukaristone

**Comment by baoea** [comment]
@ayyubibrahimi there are some `agency_slug` with triple dash such as `lsu---alexandria-university-pd`  (LSU - Alexandria University Police Department). For the best practice, I believe we should keep it to be single dash `lsu-alexandria-university-pd`. Thanks!

**Comment by ayyubibrahimi** [comment]
@baoea agreed. This issue was just addressed in the latest commit to wrgl.

**Comment by baoea** [comment]
@ayyubibrahimi we check the data and see the `dot` in `agency_slug` (for example `st.-bernard-so`). It should be `st-bernard-so`.
Thanks!

**Comment by ayyubibrahimi** [comment]
Corrected in the most recent commit. Also, note that the slugs and names have changed for some of the agencies associated with `da` or the `District Attorney's Office`

**Comment by huy-d-nguyen** [comment]
Hi @ayyubibrahimi , we think that there is one missing row in the department table, currently we only have 2 departments related to the  `feliciana`: west and east(see image)
<img width="615" alt="image" src="https://user-images.githubusercontent.com/60863227/190299196-61c00e8e-93a7-4db4-9da5-8420702e7177.png">
But from the personnel table, we have three, and the missing one is `Feliciana SO`. 
<img width="948" alt="image" src="https://user-images.githubusercontent.com/60863227/190299401-73d8995e-e6c6-4320-94f0-fbc58dddef4f.png">
Please take a look on it, and furthermore please let us know when you update all `agency` column type of all tables to slug field.

**Comment by ayyubibrahimi** [comment]
@huy-d-nguyen - I'm currently in the process of changing all of the `agency` column type to represent the slug field in PR #346. This issue should be fixed after this PR is merged. I will keep you updated.

**Comment by ayyubibrahimi** [comment]
The `agency` column type in the `event` table @baoea  and `personnel` table @huy-d-nguyen  is now the same as the `agency_slug` from the `department` table

**Comment by huy-d-nguyen** [comment]
We also see the updated slugs of the table `allegation`, `appeal-hearing`, `use-of-force`. Are those table use the new `agency_slug` too?

**Comment by ayyubibrahimi** [comment]
Yes. To avoid redundant departments again, the values have been changed for all relevant tables.

**Comment by baoea** [comment]
@ayyubibrahimi we have issue with some departments that have `department` in names. In `agency` table, we have `jefferson-fire-department`, but it is `jefferson-fire-dept` in `personnel` table. Please check it out. Thanks!

![image](https://user-images.githubusercontent.com/93108394/190545626-88cacd2f-a066-48a2-958c-f4bc5b28bf7d.png)
![image](https://user-images.githubusercontent.com/93108394/190545655-aa720cb8-4369-440c-8ddd-bea0007cbea1.png)

**Comment by baoea** [comment]
@ayyubibrahimi 

![image](https://user-images.githubusercontent.com/93108394/190547663-a36a9d13-672c-4ae1-875e-ad0e10305bd3.png)

This should be `new-orleans-levee-pd`

**Comment by ayyubibrahimi** [comment]
hey @baoea 

1) The `personnel` table should now have `jefferson-fire-department` as it's value
2) `orleans-levee-pd`, as it is in the allegations table is actually the correct slug, therefore, I've made a change to the `department` table to have `orleans-levee-pd` as the `agency_slug`

**Comment by baoea** [comment]
@ayyubibrahimi These agencies below in `personnel` table that cannot be looked up in `agency` table
Please check and fix them. Thank you very much in advance!

east-baton-rouge-so
medical-center-of-la-no
office-of-youth-dev,-dept-of-corrections
louisiana-dept-of-justice
Abbeville PD
delgado-cc-university-pd
dept-of-public-safety
lafayette-city-park-rec
orleans-da
orleans-constable
louisiana-dept-of-environmental-quality
jefferson-1st-court
housing-authority-of-no
louisiana-dept-motor-vehicle
new-orleans-levee-pd
orleans-civil-so
louisiana-dept-of-revenue
louisiana-dept-of-transportation-pd
retired
baton-rouge-cc-university-pd
bossier-cc-university-pd
city-park-pd-no
jefferson-ist-court
dept-of-health-hospitals
louisiana-dept-of-insurance
14th-jdc-da
lsuhsc-no-university-pd
delta-cc-pd-university-pd
delhl-pd
5th-jdc-district-attorneys-office
d-reserve
pox-la-pd
xc-stoula-pd
29th-jdc-da
louisiana-tech
mandbville-pd
xavier
neveans-pd
orleans-coroners-office
42nd-jdc
feliciana-so
hannon-pd
pearl-river-pd-deceased
houmap
orleans-criminal-district-court
houma
25th-jdc
river-bridge-pd
40th-jdc
se-la-flood-protection-auth-e
3rd-jdc
scort-pd
probationparoleadult-fulltime
nunez-cc-university-pd
houma-pl
ponciatoula-pd
4th-jdc
jeeferson-so
louisiana-house-of-rep-sergeant-at-arms
oaparish-so
nc-oula-pd
loyola

**Comment by ayyubibrahimi** [comment]
@baoea All tables, including the `department` table have been updated. Please review. Apologies for the lack of standardization as there are still some kinks in the OCR.

**Comment by baoea** [comment]
@ayyubibrahimi These slugs in `personnel` table still are not working. Please check. Thanks!

Abbeville PD
lafayette-city-park-rec
office-of-youth-dev,-department-of-corrections
new-orleans-criminal-court
lsuhsc-no-university-pd
new-orleans-civil-so

**Comment by baoea** [comment]
@ayyubibrahimi This a good practice for slug. You can find more in here https://docs.djangoproject.com/en/4.1/ref/utils/#django.utils.text.slugify

<img width="680" alt="image" src="https://user-images.githubusercontent.com/93108394/190603697-7411084e-c1b1-40c1-a08c-fed58dea6128.png">

Also, the comma should not be in slugs because we use slugs for urls

**Comment by ayyubibrahimi** [comment]
@baoea Thanks for the advice! I've addressed the issues above. Please review and let me know if there are any issues.

**Comment by baoea** [comment]
@ayyubibrahimi we got some `slugs` that are in `officer` and `event` tables, but not in `agency`. Please check. Thanks!

broussard-fd
vermilion-o
my-shal
gretnapld
sthelena-so
ng-oaparish-so

**Comment by ayyubibrahimi** [comment]
Looks like bad OCR again. Apologies. I'll fix.

**Comment by ayyubibrahimi** [comment]
@baoea is the `officer` table the same as the `personnel` table?

**Comment by huy-d-nguyen** [comment]
Yes, we refer the `officer` is the `personnel` table.

**Comment by baoea** [comment]
@ayyubibrahimi we still have problem with `vermilion-o` and `broussard-fire-department`. Cannot find them in `agency` table

**Comment by ayyubibrahimi** [comment]
@baoea still working on it. The bug isn't clear on my side. I'll let you know when it's fixed.

**Comment by ayyubibrahimi** [comment]
@baoea looks good. Please review.

**Comment by baoea** [comment]
The slug issue is ok now. Thank you!
